returnn/util/basic.py

```diff
@@ -965,6 +965,8 @@ def better_repr(o):
     return "set([\n%s])" % "".join(map(lambda v: better_repr(v) + ",\n", o))
   if isinstance(o, float) and (math.isnan(o) or math.isinf(o)):
     return "float('%s')" % repr(o)
+  if isinstance(o, np.ndarray):
+    return better_repr(o.tolist())
```

Comment:
This is not exactly correct. It should have `numpy.array(...)` around. The fallback code below would already handle it correctly. Why do you need this?

Reply:
The default `repr` doesn't add numpy in front:
```
>>> repr(numpy.array(0.0, dtype=numpy.float32))
'array(0., dtype=float32)'
```
I.e. you need to `from numpy import array, float32` to eval the expression. What makes parsing the output even more error prone.
But I agree that the current solution is not correct in `better_repr`, so I would remove it.

Reply:
The better solution would be to add another output file format (e.g. json) that does not rely on `repr` and would be more suitable to be parsed by other applications.

Reply:
Well, `better_repr` is to create a repr in Python format. For any other format, you would not use `better_repr`.

Reply:
Also note that the representation of a numpy array does not contain all the data if the array is large.
For the eval task I added `with numpy.printoptions(threshold=sys.maxsize):` to print all scores.